src/uterm_input_uxkb.c

```diff
@@ -158,6 +163,47 @@ int uxkb_desc_init(struct uterm_input *input,
 			   model, layout, variant, options);
 	}
 
+	locale = getenv("LC_ALL");
+	if (!locale || !*locale) {
+		locale = getenv("LC_CTYPE");
+	}
+	if (!locale || !*locale) {
+		locale = getenv("LANG");
+	}
+	if (!locale || !*locale) {
+		locale = "C";
+	}
+
+	if (compose_file && *compose_file) {
+		compose_filp = fopen(compose_file, "rb");
+		if (compose_filp) {
+			input->compose_table = xkb_compose_table_new_from_file(
+						input->ctx,
+						compose_filp,
+						locale,
+						XKB_COMPOSE_FORMAT_TEXT_V1,
+						0);
+
+			if (!input->compose_table) {
+				llog_warn(input, "cannot load compose file, "
+					  "reverting to system default");
+			}
+
+			fclose(compose_filp);
+		} else {
+			llog_warn(input, "cannot open compose file, "
+				  "reverting to system default");
+		}
+
+	}
```

Comment:
I have moved the code. There is an old comment at https://github.com/Aetf/kmscon/pull/36/files#diff-838bd2bd14044fd41a83a1a52f1ad48e81761bc017ef4e4bdb118cd230f674fdR725-R727 mentioning a switch to mmap. This is possible now (https://xkbcommon.org/doc/current/group__keymap.html#ga19e1a8824c7c51bf5ac8a8cb8b7d40d2) and it would get rid of the extra `compose_file_len` argument.